stock_quant_merge/models/stock.py

```diff
@@ -34,7 +40,8 @@ def merge_stock_quants(self):
                 for quant in quants:
                     if (self._get_latest_move(quant2merge) ==
                             self._get_latest_move(quant)):
-                        quant2merge.sudo().qty += quant.qty
+                        quant2merge.sudo().write(
+                            self._get_merged_vals(quant2merge, quant))
                         cost += quant.cost
                         cont += 1
```

Comment:
If you're going to merge all the possible quants, some will have different costs, so you should be doing a weighted average like the standard does in the dated valuation (it's changed recently: https://github.com/odoo/odoo/commit/1d7c12022dcffce4792b9eec85706e19338cfa9f)